akka-actor/src/main/scala/akka/pattern/HandleBackoff.scala

```diff
@@ -32,6 +33,15 @@ import akka.annotation.InternalApi
     child = Some(context.watch(context.actorOf(childProps, childName)))
   }
 
+  protected def getOrCreateHandler(p: Props): ActorRef = handler match {
+    case Some(h) =>
+      h
+    case None =>
+      val h = context.actorOf(p)
```

Comment:
so the handler is started on demand, but never stopped?

I wonder if an alternative would be to be able to pass in an `ActorRef` instead of a `Props`. Kind of a way to replace the `deadLetters` ActorRef?

Reply:
It could be stopped once the child is restarted again.

Just accepting an actorRef would be the most light-weight solution and provide the same functionality.

Reply:
Pro/Cons:

With `Props` we can start handling actor only when we need it (assuming it's an exceptional case) but each supervisor would start its own.
With `ActorRef` the API user can share the handling actor across all supervised actors if desired, but would have to be eager instantiated.

Reply:
I've pushed another update just using `ActorRef`. Adds the least complexity for special cases.